beacon-chain/attestation/service.go

```diff
@@ -0,0 +1,110 @@
+// Package attestation defines the life-cycle and status of single and aggregated attestation.
+package attestation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/prysmaticlabs/prysm/beacon-chain/types"
+	"github.com/sirupsen/logrus"
+)
+
+var log = logrus.WithField("prefix", "attestation")
+
+// Service represents a service that handles the internal
+// logic of managing aggregated attestation.
+type Service struct {
+	ctx           context.Context
+	cancel        context.CancelFunc
+	handler       *Handler
+	broadcastFeed *event.Feed
+	broadcastChan chan *types.Attestation
+	incomingFeed  *event.Feed
+	incomingChan  chan *types.Attestation
+}
+
+// Config options for the service.
+type Config struct {
+	Handler                 *Handler
+	ReceiveAttestationBuf   int
+	BroadcastAttestationBuf int
+}
+
+// NewService instantiates a new service instance that will
```

Comment:
We have too many NewServices in our repo that makes grep'ing a bit hard. I know all of these are services but for the sake of differentiation it could be better to be more semantic here. Maybe AttestationService?

Reply:
Gonna name it `NewAttestService` 👌